src/test/java/uk/gov/pay/ledger/transaction/model/CsvTransactionFactoryTest.java

```diff
@@ -403,6 +403,46 @@ void toMapShouldIncludeShouldInclude3DSecureRequired() {
         assertThat(csvDataMap.get("3-D Secure Required"), is(true));
     }
 
+    @Test
+    void toMapShouldReturnMapWithNullValueFor3DSecureRequiredWhenRequires3dsIsFalse() {
+
+        TransactionEntity transactionEntity = aTransactionFixture()
+                .withState(TransactionState.FAILED_REJECTED)
+                .withTransactionType(TransactionType.PAYMENT.name())
+                .withAmount(100L)
+                .withGatewayTransactionId("gateway-transaction-id")
+                .withCreatedDate(ZonedDateTime.parse("2018-03-12T16:25:01.123456Z"))
+                .withCorporateCardSurcharge(23)
+                .withCardBrandLabel("Visa")
+                .withDefaultCardDetails()
+                .withFee(5L)
+                .withTotalAmount(123L)
+                .withRequires3ds(false)
+                .withVersion3ds("2.0.1")
+                .withDefaultTransactionDetails()
+                .toEntity();
+
+        Map<String, Object> csvDataMap = csvTransactionFactory.toMap(transactionEntity);
+
+        assertPaymentDetails(csvDataMap, transactionEntity);
+
+        assertThat(csvDataMap.get("Amount"), is("1.00"));
+        assertThat(csvDataMap.get("GOV.UK Payment ID"), is(transactionEntity.getExternalId()));
+        assertThat(csvDataMap.get("Provider ID"), is(transactionEntity.getGatewayTransactionId()));
+        assertThat(csvDataMap.get("State"), is("Declined"));
+        assertThat(csvDataMap.get("Finished"), is(true));
+        assertThat(csvDataMap.get("Fee"), is("0.05"));
+        assertThat(csvDataMap.get("Net"), is("0.00"));
+        assertThat(csvDataMap.get("Error Code"), is("P0010"));
+        assertThat(csvDataMap.get("Error Message"), is("Payment method rejected"));
+        assertThat(csvDataMap.get("Date Created"), is("12 Mar 2018"));
+        assertThat(csvDataMap.get("Time Created"), is("16:25:01"));
+        assertThat(csvDataMap.get("Corporate Card Surcharge"), is("0.23"));
+        assertThat(csvDataMap.get("Total Amount"), is("0.00"));
+        assertThat(csvDataMap.get("Payment Provider"), is("sandbox"));
+        assertThat(csvDataMap.get("3-D Secure Required"), is(nullValue()));
+    }
```

Comment:
Perhaps add a variant of this where it’s explicitly null and also rename `toMapShouldIncludeShouldInclude3DSecureRequired()` (which is wrong already!) to `toMapShouldInclude3DSecureRequiredWhenTrue()` or something.